src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Tests/IntegrationTests/TestServiceTests.cs

```diff
@@ -0,0 +1,195 @@
+            var startDate = DateTime.Now;
+            var wrongAnswer = "Wrong";
+            using (var context = new ApplicationDbContext(this.ContextOptions))
+            {
+                context.Tests.Add(test);
+                context.SaveChanges();
+            }
+
+            using (var context = new ApplicationDbContext(this.ContextOptions))
+            {
+                var testService = PrepareSUT(context);
+
+                testService.AddTestWrongAnswer(userId, test.Id, wrongAnswer, startDate);
+            }
+
+            using (var context = new ApplicationDbContext(this.ContextOptions))
+            {
+                var result = context.UserTestWrongAnswers.SingleOrDefault(t => t.UserId == userId && t.TestId == test.Id);
+                result.Should().NotBe(null);
+                result.Answer.Should().Be(wrongAnswer);
+            }
+        }
+
+        [Fact]
+        public void Gets_user_correct_answer_by_testId()
+        {
+            var test = GetTest();
+            var userId = "c611530e-bebd-41a9-ace2-951550edbfa0";
+            var userCorrectAnswer = new UserTestCorrectAnswer()
+            {
+                Test = test,
+                UserId = userId,
+            };
+
+            using (var context = new ApplicationDbContext(this.ContextOptions))
+            {
+                context.UserTestCorrectAnswers.Add(userCorrectAnswer);
+                context.SaveChanges();
+            }
+
+            using (var context = new ApplicationDbContext(this.ContextOptions))
+            {
+                var testService = PrepareSUT(context);
+
+                var result = testService.GetAnswerByTestId(test.Id);
+
+                result.Should().BeOfType<UserTestCorrectAnswerDto>();
+                result.TestId.Should().Be(test.Id);
+                result.UserId.Should().Be(userId);
+            }
+        }
+
+        [Fact]
+        public void Gets_user_correct_answer_by_userId()
+        {
+            var test = GetTest();
+            var userId = "c611530e-bebd-41a9-ace2-951550edbfa0";
+            var userCorrectAnswer = new UserTestCorrectAnswer()
+            {
+                Test = test,
+                UserId = userId,
+            };
+
+            using (var context = new ApplicationDbContext(this.ContextOptions))
+            {
+                context.UserTestCorrectAnswers.Add(userCorrectAnswer);
+                context.SaveChanges();
+            }
+
+            using (var context = new ApplicationDbContext(this.ContextOptions))
+            {
+                var testService = PrepareSUT(context);
+
+                var result = testService.GetAnswerByTestId(test.Id);
+
+                result.Should().BeOfType<UserTestCorrectAnswerDto>();
+                result.TestId.Should().Be(test.Id);
+                result.UserId.Should().Be(userId);
+            }
+        }
+
+        [Fact]
+        public void Has_user_answered_test()
```

Comment:
should be `Has_user_answered_test_returns_true_when_answered`
and there should be second test for false:
should be `Has_user_answered_test_returns_false_when_not_answered`